wp/lib/bap_wp/src/constraint.ml

```diff
@@ -125,7 +143,7 @@ let get_refuted_goals_and_paths (constr : t) (solver : Z3.Solver.solver)
     match constr with
     | Goal g ->
       let goal_val = Expr.substitute g.goal_val olds news in
-      let goal_res = Option.value_exn (Z3.Model.eval model goal_val true) in
+      let goal_res = Option.value_exn (Model.eval model goal_val true) in
```

Comment:
I wonder if we should implement `Option.value_exn (Model.eval m v true)` since we call it so much, and we don't get a useful error if this fails currently...

Reply:
Probably a good idea. I just grepped and found at least 7 cases where we called `Option.value_exn (Model.eval m v true)`